streams/src/main/java/org/apache/kafka/streams/processor/internals/namedtopology/RemoveNamedTopologyResult.java

```diff
@@ -52,34 +60,54 @@ public RemoveNamedTopologyResult(final KafkaFuture<Void> removeTopologyFuture) {
      * successfully completed their corresponding {@link KafkaFuture}.
      */
     public final KafkaFuture<Void> all() {
-        final KafkaFutureImpl<Void> result = new KafkaFutureImpl<>();
+        if (resetOffsetsFuture == null) {
+            return removeTopologyFuture;
+        } else {
+            return resetOffsetsFuture;
```

Comment:
Basically we now have the caller thread perform the offset reset and block on it when it goes to call `get()` on the future returned by `RemoveNamedTopologyResult#all` (or `#resetOffsetsFuture`)